src/Workspaces/CSharp/Portable/Formatting/Rules/BaseFormattingRule.cs

```diff
@@ -226,14 +227,32 @@ protected void AddBraceSuppressOperations(List<SuppressOperation> list, SyntaxNo
 
                 if (SomeParentHasMissingCloseBrace(node.Parent))
                 {
-                    endBrace = endBrace.GetPreviousToken();
+                    if (node.IsKind(SyntaxKind.Block) && ((BlockSyntax)node).Statements.Count >= 1)
```

Comment:
Up to you.  I personally prefer it.  Instead of parsing a boolean expression, my brain just needs to look at a member access expression.  Here you care that node is a `BlockSyntax` but only so you can get the first statement.  I normally feel that using the conditional operator make the intent clearer.  Also, then you don't need to cast twice:
```C#
var blockSyntaxNode = node as BlockSyntax
if (blockSyntaxNode?.Statements.Count >= 1)
{
    var firstStatement = blockSyntaxNode.Statements[0];
    if (FormattingRangeHelper.AreTwoTokensOnSameLine(firstTokenOfNode, firstStatement.GetFirstToken()))
    {
        endToken = firstStatement.GetLastToken();
    }
}
```

Anyways, its a style comment.  I don't think this blocks checkin.